src/webui/abstractwebapplication.cpp

```diff
@@ -307,10 +308,12 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (Utils::Net::isLANAddress(env_.clientAddress))
```

Comment:
Thinking through this more, the only way I see to solve the IPv6 issue is to implement the ACL you mentioned earlier. I don't think this method will be easy for newbies but for more advanced users it'll be great.